pgcli/pgexecute.py

```diff
@@ -189,6 +190,11 @@ def run(self, statement):
                 _logger.debug('Successfully switched to DB: %r', dbname)
                 yield (None, None, None, 'You are now connected to database "%s" as '
                         'user "%s"' % (self.dbname, self.user))
+            elif iospecial.editor_command(command):
+                tokens = sql.split(' ', 2)
+                filename = tokens[1] if len(tokens) > 1 and tokens[1] else None
+                for result in iospecial.open_external_editor(filename=filename):
```

Comment:
Can you pass in the statement to this function as an arg? That way a partially typed query such as:

```
SELECT * FROM \e
```

will have `SELECT * FROM \e` passed in which will be used to populate the partial buffer.